q2doc/usage/domain.py

```diff
@@ -0,0 +1,23 @@
+from sphinx.domains import Domain
+
+from q2cli.core.usage import CLIUsage
+from qiime2.plugins import ArtifactAPIUsage
+from qiime2.sdk import usage as usage
+
+from .directive import UsageDirective
+
+
+class QIIME2Domain(Domain):
+    name = 'q2'
+    label = 'QIIME 2 Domain'
+
+    directives = {'usage': UsageDirective}
+
+    initial_data = {
+        'drivers': {
+            'dia_use': usage.DiagnosticUsage,
+            'exc_use': usage.ExecutionUsage,
+            'cli_use': CLIUsage,
+            'art_use': ArtifactAPIUsage,
+        },
+    }
```

Comment:
Not sure if it makes more sense to stick this on the `env` object [as I had originally done](https://github.com/qiime2/sphinx-ext-qiime2/blob/83e5836f01b4c795e1087eb362a684d091d1bb60/q2doc/usage/directive.py#L25-L31) after removing the `MetaUsage(Enum)` driver.